src/gt4py/next/iterator/ir.py

```diff
@@ -203,6 +203,7 @@ class FencilDefinition(Node, ValidatedSymbolTableTrait):
     function_definitions: List[FunctionDefinition]
     params: List[Sym]
     closures: List[StencilClosure]
+    implicit_domain: bool = False
```

Comment:
This was added to make it possible for backends to decide whether their `decorate` step needs to generate size arguments. I'm glad to talk about better ways, though preferably in terms of a reference implementation and as a topic for a follow-up PR.